pkg/daemon/gpsd.go

```diff
@@ -112,13 +116,89 @@ func (g *gpsd) cmdRun(stdoutToSocket bool) {
 			time.Sleep(1 * time.Second)
 			err = g.cmd.Start() // this is asynchronous call,
 			if err != nil {
-				glog.Errorf("cmdRun() error starting %s: %v", g.Name(), err)
+				glog.Errorf("CmdRun() error starting %s: %v", g.Name(), err)
 			}
 		}
 		<-done // goroutine is done
 		err = g.cmd.Wait()
 		if err != nil {
-			glog.Errorf("cmdRun() error waiting for %s: %v", g.Name(), err)
+			glog.Errorf("CmdRun() error waiting for %s: %v", g.Name(), err)
 		}
 	}
 }
+
+// MonitorGNSSEvents ... monitor gnss events
+func (g *gpsd) monitorGNSSEvents(processCfg config.ProcessConfig, pluginName string) error {
+	//done := make(chan struct{}) // Done setting up logging.  Go ahead and wait for process
+	// var currentNavStatus int64
+	g.processConfig = processCfg
+	var err error
+	//currentNavStatus = 0
+	glog.Infof("Starting GNSS Monitoring for plugin  %s...", pluginName)
+
+	var ublx *ublox.UBlox
+	g.state = FREERUN
+retry:
+	if ublx, err = ublox.NewUblox(); err != nil {
+		glog.Errorf("failed to initialize GNSS monitoring via ublox %s", err)
+		time.Sleep(10 * time.Second)
+		goto retry
+	} else {
+		//TODO: monitor on 1PPS  events trigger
+		ticker := time.NewTicker(1 * time.Second)
+		for {
+			select {
+			case <-ticker.C:
+				// do stuff
+				nStatus, errs := ublx.NavStatus()
+				offsetS, err2 := ublx.GetNavOffset()
+				if errs == nil && err2 == nil {
+					//calculate PTP states
+					g.offset, _ = strconv.ParseInt(offsetS, 10, 64)
+					if nStatus > 3 && g.isOffsetInRange() {
+						g.state = event.PTP_LOCKED
+					} else {
+						g.state = event.PTP_FREERUN
+					}
+					processCfg.EventChannel <- event.EventChannel{
```

Comment:
should we save the previous g.state and send out events only if g.state is changed?